job/server/src/main/java/alluxio/master/job/metrics/DistributedCmdMetrics.java

```diff
@@ -0,0 +1,231 @@
+        }
+    }
+
+    // Increment by a fileSize
+    public static void incrementFileSize(String jobType, long fileSize) {
+        switch (jobType) {
+            case LoadConfig.NAME:
+                JOB_DISTRIBUTED_LOAD_FILE_SIZE.inc(fileSize);
+                JOB_DISTRIBUTED_LOAD_RATE.mark(fileSize);
+                break;
+            case MigrateConfig.NAME:
+                MIGRATE_JOB_FILE_SIZE.inc(fileSize);
+                break;
+            case PersistConfig.NAME:
+                ASYNC_PERSIST_FILE_SIZE.inc(fileSize);
+                break;
+            default:
+                LOG.warn("JobType does not belong to Load, Migrate and Persist");
+                break;
+        }
+    }
+
+    //Increment failed job count
+    public static void incrementForFailStatus(String jobType) {
+        switch (jobType) {
+            case LoadConfig.NAME:
+                JOB_DISTRIBUTED_LOAD_FAIL.inc();
+                break;
+            case MigrateConfig.NAME:
+                MIGRATE_JOB_FAIL.inc();
+                break;
+            case PersistConfig.NAME:
+                ASYNC_PERSIST_FAIL.inc();
+                break;
+            default:
+                LOG.warn("JobType does not belong to Load, Migrate and Persist");
+                break;
+        }
+    }
+
+    //Increment cancelled job count
+    public static void incrementForCancelStatus(String jobType) {
+        switch (jobType) {
+            case LoadConfig.NAME:
+                JOB_DISTRIBUTED_LOAD_CANCEL.inc();
+                break;
+            case MigrateConfig.NAME:
+                MIGRATE_JOB_CANCEL.inc();
+                break;
+            case PersistConfig.NAME:
+                ASYNC_PERSIST_CANCEL.inc();
+                break;
+            default:
+                LOG.warn("JobType does not belong to Load, Migrate and Persist");
+                break;
+        }
+    }
+
+    // Increment for non-batch config for complete status
+    public static void incrementForCompleteStatusWithRetry(JobConfig config, FileSystem fileSystem, int RETRY_COUNTS) {
+        String jobType = config.getName();
+        String filePath = getFilePathForNonBatchConfig(config);
+        incrementOperationCount(jobType, 1);
+        incrementFileCount(jobType, 1);
+
+        for (int i = 0; i < RETRY_COUNTS; i++) {
+            try {
+                long fileSize = fileSystem.getStatus(new AlluxioURI(filePath)).getLength();
+                incrementFileSize(jobType, fileSize);
+                break;
+            } catch (IOException | AlluxioException e) {
+                i++;
+                LOG.warn("Retry getStatus for URI {} for {}-th time, {}", filePath, i, Arrays.toString(e.getStackTrace()));
+            }
+        }
+    }
+
+    // Get file path based on each non-batch job config
+    private static String getFilePathForNonBatchConfig(JobConfig config) {
+        String path = null;
+        if (config.getName().equals(LoadConfig.NAME)) {
```

Comment:
why not use instanceof? And I think in this case we don't really need to change the source parameter for MigrateConfig

Reply:
done